pkg/querier/tripperware/instantquery/instant_query.go

```diff
@@ -120,8 +141,14 @@ func (instantQueryCodec) EncodeRequest(ctx context.Context, r tripperware.Reques
 		}
 	}
 
-	// Always ask gzip to the querier
-	h.Set("Accept-Encoding", "gzip")
+	if c.compression == queryrange.GzipCompression {
+		h.Set("Accept-Encoding", string(c.compression))
+	}
+	if c.enableProtobuf {
+		h.Set("Accept", "application/x-protobuf")
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept#syntax
I suggest we add json type as well but we put protobuf first so it should choose protobuf if supported. Otherwise it can just fallback to JSON.

```
application/x-protobuf, application/json
```

Reply:
json `Accept` header is always added now and we only add protobuf before it if the config is enabled